src/pages/workspace/companyCards/WorkspaceCompanyCardDetailsPage.tsx

```diff
@@ -0,0 +1,197 @@
+    const connectedIntegration = getConnectedIntegration(policy, accountingIntegrations) ?? connectionSyncProgress?.connectionName;
+
+    const [personalDetails] = useOnyx(ONYXKEYS.PERSONAL_DETAILS_LIST);
+    const [allBankCards] = useOnyx(`${ONYXKEYS.COLLECTION.WORKSPACE_CARDS_LIST}${workspaceAccountID}_${bank}`);
+    const card = allBankCards?.[cardID];
+
+    const cardholder = personalDetails?.[card?.accountID ?? -1];
+    const displayName = PersonalDetailsUtils.getDisplayNameOrDefault(cardholder);
+    const exportMenuItem = getExportMenuItem(connectedIntegration, policyID, translate, policy, card);
+
+    const unassignCard = () => {
+        setIsUnassignModalVisible(false);
+        Policy.unassignWorkspaceCompanyCard(workspaceAccountID, cardID, bank);
+        Navigation.goBack();
+    };
+
+    const updateCard = () => {
+        Policy.updateWorkspaceCompanyCard(workspaceAccountID, cardID, bank);
+    };
+
+    return (
+        <AccessOrNotFoundWrapper
+            policyID={policyID}
+            featureName={CONST.POLICY.MORE_FEATURES.ARE_COMPANY_CARDS_ENABLED}
+        >
+            <ScreenWrapper
+                includeSafeAreaPaddingBottom={false}
+                testID={WorkspaceCompanyCardDetailsPage.displayName}
+            >
+                {({safeAreaPaddingBottomStyle}) => (
+                    <>
+                        <HeaderWithBackButton
+                            title={translate('workspace.moreFeatures.companyCards.cardDetails')}
+                            onBackButtonPress={() => Navigation.goBack(backTo)}
+                        />
+                        <ScrollView contentContainerStyle={safeAreaPaddingBottomStyle}>
+                            <View style={[styles.walletCard, styles.mb3]}>
+                                <ImageSVG
+                                    contentFit="contain"
+                                    src={CardUtils.getCardDetailsImage(card?.bank ?? '')}
+                                    pointerEvents="none"
+                                    height={variables.cardPreviewHeight}
+                                    width={variables.cardPreviewWidth}
+                                />
+                            </View>
+
+                            <MenuItem
+                                label={translate('workspace.moreFeatures.companyCards.cardholder')}
+                                title={displayName}
+                                icon={cardholder?.avatar ?? FallbackAvatar}
+                                iconType={CONST.ICON_TYPE_AVATAR}
+                                description={cardholder?.login}
+                                interactive={false}
+                            />
+                            <MenuItemWithTopDescription
+                                description={translate('workspace.moreFeatures.companyCards.cardNumber')}
+                                title={CardUtils.maskCard(card?.lastFourPAN)}
```

Comment:
It caused [this issue](https://github.com/Expensify/App/issues/51912), we need to mask the card name to display 6 first numbers and 4 last numbers